internal/local/structure_v2_test.go

```diff
@@ -66,3 +68,259 @@ func TestParseDataSources(t *testing.T) {
 		}}, dataSources)
 	})
 }
+
+func TestWriteFunctionsV2(t *testing.T) {
+	tmpDir, cleanupTmpDir, err := u.NewTempDir("")
+	assert.Nil(t, err)
+	defer cleanupTmpDir()
+
+	t.Run("should write functions to disk", func(t *testing.T) {
+		data := &FunctionsStructure{
+			Configs: []map[string]interface{}{
+				{
+					"name":    "test",
+					"private": true,
+				},
+			},
+			Sources: map[string]string{
+				"test.json": "exports = function(){\n  console.log('Hello World!');\n};",
```

Comment:
[nit] the extension here should be `.js`

and it might be worth adding an entry here where:

`"nested/test.js": "exports = function(){\n  console.log('Hello World!');\n}"`

and ensure we write that to the proper directory

Reply:
good catch!